test/rules/use_super_initializers_test.dart

```diff
@@ -0,0 +1,205 @@
+}
+''', [
+      lint('use_super_initializers', 79, 17),
+    ]);
+  }
+
+  test_named_middle() async {
+    await assertDiagnostics(r'''
+class A {
+  A({int? x, int? y, int? z});
+}
+class B extends A {
+  B({int? x, int? y, int? z}) : super(x: x, y: y, z: z);
+}
+''', [
+      lint('use_super_initializers', 95, 23),
+    ]);
+  }
+
+  test_named_only() async {
+    await assertDiagnostics(r'''
+class A {
+  A({int? x});
+}
+class B extends A {
+  B({int? x}) : super(x: x);
+}
+''', [
+      lint('use_super_initializers', 63, 11),
+    ]);
+  }
+
+  test_no_lint_named_noSuperInvocation() async {
+    await assertNoDiagnostics(r'''
+class A {
+  A({int x = 0});
+}
+class B extends A {
+  B({int x = 1});
+}
+''');
+  }
+
+  test_no_lint_named_notGenerative() async {
+    await assertNoDiagnostics(r'''
+class A {
+  A({required int x});
+}
+class B extends A {
+  static List<B> instances = [];
+  factory B({required int x}) => instances[x];
+}
+''');
+  }
+
+  test_no_lint_named_notPassed_unreferenced() async {
+    await assertNoDiagnostics(r'''
+class A {
+  A({int x = 0});
+}
+class B extends A {
+  B({int x = 0}) : super(x: 0);
+}
+''');
+  }
+
+  test_no_lint_named_notPassed_usedInExpression() async {
+    await assertNoDiagnostics(r'''
+class A {
+  A({String x = ''});
+}
+class B extends A {
+  B({required Object x}) : super(x: x.toString());
+}
+''');
+  }
+
+  test_no_lint_requiredPositional_noSuperInvocation() async {
+    await assertNoDiagnostics(r'''
+class A {
+  A();
+}
+class B extends A {
+  B(int x);
+}
+''');
+  }
+
+  test_no_lint_requiredPositional_notGenerative() async {
+    await assertNoDiagnostics(r'''
+class A {
+  A(int x);
+}
+class B extends A {
+  static List<B> instances = [];
+  factory B(int x) => instances[x];
+}
+''');
+  }
+
+  test_no_lint_requiredPositional_notPassed_unreferenced() async {
+    await assertNoDiagnostics(r'''
+class A {
+  A(int x);
+}
+class B extends A {
+  B(int x) : super(0);
+}
+''');
+  }
+
+  test_no_lint_requiredPositional_notPassed_usedInExpression() async {
+    await assertNoDiagnostics(r'''
+class A {
+  A(String x);
+}
+class B extends A {
+  B(Object x) : super(x.toString());
+}
+''');
+  }
+
+  test_optionalPositional_singleSuperParameter_only() async {
+    await assertDiagnostics(r'''
+class A {
+  A(int x);
+}
+class B extends A {
+  B([int x = 0]) : super(x);
+}
+''', [
+      lint('use_super_initializers', 63, 8),
+    ]);
+  }
+
+  test_requiredPositional_mixedSuperParameters_first() async {
+    await assertDiagnostics(r'''
+class A {
+  A(int x, {int? y});
+}
+class B extends A {
+  B(int x, int y) : super(x, y: y);
+}
+''', [
+      lint('use_super_initializers', 74, 14),
+    ]);
+  }
+
+  test_requiredPositional_mixedSuperParameters_last() async {
+    await assertDiagnostics(r'''
+class A {
+  A(int x, {int? y});
+}
+class B extends A {
+  B(int y, int x) : super(x, y: y);
+}
+''', [
+      lint('use_super_initializers', 74, 14),
```

Comment:
I think that several of these tests not really testing what you want because of the fact that we're flagging the super initializer rather than individual parameters, and as a result, the tests can't determine when one parameter is linted but others that ought to be linted won't be. I think that's especially a problem here because the original test was checking that we don't offer to convert the parameter `x` in `B`s constructor. What should really be tested here is
```
class A {
  A({int? x});
}
class B extends A {
  B(int x) : super(x: x);
}
```
where there should be no lint (because a positional parameter can only be a super parameter if the corresponding parameter is also positional).

Reply:
Good catch!